doc/LibrarySplit/HowToSplit.md

```diff
@@ -0,0 +1,215 @@
+- Preserve backwards compatbility between the original Package v1.0.0 and the new Package v1.1.0
+- Move all related types
+    - Resource
+    - ResourceData
+    - ResourceCollection
+    - Any other models used by those types
+    - Extension methods
+    - Mocking helpers
+
+---
+## High level scenario
+
+For all sections in this doc we will use the same scenario where we start with
+a single package called Foo v1.0.0 containing three resources: Resource1,
+Resource2, and Resource3.  We want to split the package into two by moving
+Resource1 into a new package Bar v1.0.0.  Foo will bump to v1.1.0 and still have
+Resource2 and Resource3 inside it but will have a dependency on Bar v1.0.0 and
+type forward Resource1 into Bar.
+
+## Dependency
+
+The  new package Foo v1.1.0 will need to depend on Bar v1.0.0 in order to support
+type forwarding.  This does introduce a release dependency but will minimize the
+impact on our library consumers.
+
+Bar will be able to release independently of Foo as either beta or GA for future
+versions.
+
+Foo will also be able to release independently of Bar as either beta or GA for
+future versions.  Whenever Foo wants to ship a new version it should bump its
+dependency on Bar according to which support model it is shipping under.  If Foo
+is shipping as beta it should ship with a dependency on the latest version of Bar
+regardless of support model.  If Foo is shipping as GA it should ship with a
+dependency on the latest GA version of Bar.
+
+## Types Being Moved
+
+Each type that is moved **MUST** have an identical public API surface in the new
+package.  This includes any hidden EBN members that were put in place for
+backwards compatibility in the original package.
+
+## TypeForwardedTo
+
+To maintain binary compatibility, Foo v1.1.0 will need to use `TypeForwardedTo`.
+For organization we will create a new file called `TypesForwardedTo[NewPackage].cs`
+for each new package that Foo is splitting into.
+
+In our example a new file called `TypesForwardedToBar.cs` should be added to
+Foo v1.1.0 and this will contain a list of all types that have been moved to Bar
+v1.0.0.  This is the only package we are splitting into so its the only file we
+need to add.
+
+`TypesForwardedToBar.cs` (in Foo v1.1.0):
+```csharp
+using System.Runtime.CompilerServices;
+
+[assembly: TypeForwardedTo(typeof(Resource1))]
+[assembly: TypeForwardedTo(typeof(ResourceData1))]
+[assembly: TypeForwardedTo(typeof(Resource1Collection))]
+// Add additional forwarded dependent property-only types if they moved
+```
+
+## TypeForwardedFrom
+
+For some tooling and serializers it is helpful to know where a type was
+originally defined.  To help with this we will add the `TypeForwardedFrom`
+attribute on each type that is being moved to a new package.
+
+`Resource1.cs` (in Bar v1.0.0):
+```csharp
+using System.Runtime.CompilerServices;
+
+namespace Bar
+{
+    [TypeForwardedFrom("Foo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=[key]")]
```

Comment:
oof.   Bidirectional is messy.

Reply:
Yeah without this any reflection or binary serialization use cases can break.